### ISSUE TITLE
'Created by Spellslinger' incorrectly displayed for Spectators

### ISSUE THREAD

**Issue by Patashu** [HearthSim/hs-bugs#49]
Last verified in: Patch 3.0.0.9786 (TGT patch)

If you are spectating someone who gets a spell created by Spellslinger, the text displayed for it is incorrect.

http://imgur.com/ioEPGN6

**Comment by Patashu** [comment]
This might not be a spectator bug, this has the same visual glitch but it's not a spectator:

https://www.reddit.com/r/hearthstone/comments/3j8nl9/arena_bug_paladin_with_mage_secrets/

**Comment by Patashu** [comment]
Another report: https://www.reddit.com/r/hearthstone/comments/3jk0rq/found_a_minor_glitch_while_spectating/

**Comment by culinko** [comment]
https://www.reddit.com/r/hearthstone/comments/3wdhzt/new_bug_created_by/

This bug might be related to reconnecting or spectating after the card was added into the hand.

**Comment by Patashu** [comment]
https://www.reddit.com/r/hearthstone/comments/3wdhzt/new_bug_created_by/ Another report

**Comment by culinko** [comment]
Report from 4.2: https://www.reddit.com/r/hearthstone/comments/4c4p59/hearthstone_glitch_reveals_blizzards_programming/

**Comment by Patashu** [comment]
Confirmed in 4.3:

http://us.battle.net/hearthstone/en/forum/topic/20743184773

**Comment by Patashu** [comment]
Confirmed in 5.0:

http://us.battle.net/hearthstone/en/forum/topic/20744944850

**Comment by Patashu** [comment]
Confirmed in 5.0.0.103:

http://us.battle.net/hearthstone/en/forum/topic/20745264555

**Comment by culinko** [comment]
Confirmed in 6.0, looks like the entity doesn't need to be destroyed, just outside of play zone? https://www.reddit.com/r/hearthstone/comments/51h7my/bug_antonidas_dreams_of_type_invalid_sheep/

**Comment by culinko** [comment]
Also happens with Prince Malchezaar and reconnect:
* https://www.reddit.com/r/hearthstone/comments/50peu8/bug_prince_malchezaars_legendary_not_showing_its/
* https://www.reddit.com/r/hearthstone/comments/51ce4f/invalid_deck_zone_grommash_hacked_into_the_deck/
* https://www.reddit.com/r/hearthstone/comments/51k7hv/if_you_reconnect_to_a_game_where_your_opponent/

**Comment by Patashu** [comment]
Confirmed in 6.1.1:

https://www.reddit.com/r/hearthstone/comments/55b2n2/hearthstone_spectator_bug/

**Comment by jleclanche** [comment]
I don't think it has to do with the entity getting "removed", but rather the entity getting *hidden*.

Eg. when Prince Malchezaar shows up at the beginning of the game, he's revealed but then he's hidden again. So when a spectator connects, they don't have a matching cardID.

**Comment by Patashu** [comment]
Looks like it still happens:

https://twitter.com/xkeepah/status/805652408604848128

**Comment by judge2020** [comment]
Same for 7.0:
